src/QueryM/Ogmios.purs

```diff
@@ -430,18 +445,56 @@ instance Show EraSummaryParameters where
 instance DecodeAeson EraSummaryParameters where
   decodeAeson = aesonObject $ \o -> do
     epochLength <- getField o "epochLength"
-    slotLength <- getField o "slotLength"
+    slotLength <- getField o "slotLength" >>= stringifyAeson >>> parseSlotLength
     safeZone <- fromMaybe zero <$> getField o "safeZone"
     pure $ wrap { epochLength, slotLength, safeZone }
 
+slotLengthFactor :: Int
+slotLengthFactor = 1000
+
+parseSlotLength :: String -> Either JsonDecodeError SlotLength
```

Comment:
I'm not sure if I'm missing something -- but can we just store the `SlotLength` as a `Number`?

Reply:
About that, since we have this [https://github.com/input-output-hk/ouroboros-network/blob/69fc48e9b0b7f845e2cfb40e8daaae7ea11548bb/ouroboros-consensus/src/Ouroboros/Consensus/HardFork/History/EraParams.hs#L69](https://github.com/input-output-hk/ouroboros-network/blob/69fc48e9b0b7f845e2cfb40e8daaae7ea11548bb/ouroboros-consensus/src/Ouroboros/Consensus/HardFork/History/EraParams.hs#L69 ) : 

```
 When a hard fork point is still unknown, we assume that each era determines a
  "safe zone": a number of slots from the tip of the ledger in which it is
  guaranteed that the hard fork will not happen.
  > CASE (i)
  >
  > ---------------A--------------*----------?--------------------
  >                               \..../
  >                                safe
  >                                zone
  Since the hard fork will not happen in the safe zone, we can extend the use of
  the current set of era parameters past the tip into the safe zone, giving us a
  limited ability to make predictions for the future (such as converting between
  slot numbers and wallclock time).
```

And as seen in [this slack discussion](https://mlabs-corp.slack.com/archives/C02QXGRGAPK/p1658940234733219),  the `safe zone` right now is around 5 days.

If the slot length becomes incredibly large, the mechanism to determine the `safe zone` and others may begin to work with fractions of `slots`. This means we would need to handle a new unit of time that would replace `slots` in most of its use cases. So, either this won't happen or we would need to modify the code to handle the new units aniway. 

So I believe there are low changes that `SlotLenght` would become incredibly long. It maybe would grow to 10 or 100 in the following years but not much more.

Reply:
Also, becoming incredibly small is even less probable to happen since it makes little sense to use something below micro seconds (even using microseconds is senseless for the kind of task a `slot` represents).